docs/feature_pointing_device.md

```diff
@@ -909,3 +910,708 @@ In general the following two functions must be implemented in appropriate locati
 | -------------------------------------------------------------- | ------------------------------------------------------------ | ---------------------------: |
 | `pointing_device_task_auto_mouse(report_mouse_t mouse_report)` | handles target layer activation and is_active status updates | `pointing_device_task` stack |
 | `process_auto_mouse(uint16_t keycode, keyrecord_t* record)`    | Keycode processing for auto mouse                            |       `process_record` stack |
+
+---
+
+# Pointing Device Modes :id=pointing-device-modes
+
+Inspired by the work of previous trackball users that added features such as drag scroll, caret scroll, and sniping modes to their keyboards, this framework allows for easy setup and inclusion of different pointing device modes that when active will change the behaviour of a pointing device by taking it's x/y outputs and changing them into something else such as h/v for drag scrolling, key presses such as arrow keys for caret scrolling, and even just adjusting the x/y values before output.  When a pointing device mode is active it accumulates x and y outputs from a pointing device and stores it into internal x & y values, halting normal mouse x and y output (_modes can re-enable and/or modify mouse output_), these internally stored x and y values are then divided by a defined divisor resulting the modified output (_key taps, h/v, modified mouse x/y etc._). The dividing factors can be used to control sensitivity in each mode as adjusting cpi may not always be desired/possible.   
```

Comment:
As a minor spelling/grammar nitpick, the fourth line should be ..."by taking its x/y"... rather than "it's".

Reply:
Thanks for this I have re written the intro a bit and cleaned up the grammar hopefully.